src/undate/date.py

```diff
@@ -29,6 +31,144 @@ def days(self) -> int:
         return int(self.astype("datetime64[D]").astype("int"))
 
 
+@dataclass
+class UnInt:
+    """An uncertain integer intended for use with uncertain durations (:class:`UnDelta`),
+    to convey a range of possible integer values between an upper
+    and lower bound (inclusive). Supports comparison, addition and subtraction,
+    checking if a value is included in the range, and iterating over numbers
+    included in the range.
+    """
+
+    lower: int
+    upper: int
+
+    def __post_init__(self):
+        # validate that lower value is less than upper
+        if not self.lower < self.upper:
+            raise ValueError(
+                f"Lower value ({self.lower}) must be less than upper ({self.upper})"
+            )
```

Comment:
Would it make sense to allow UnInts for exact values, meaning to allow an `UnInt(1,1)`? I'm thinking of cases where maybe someone would iterate over an UnInt making the error margin smaller and smaller until it reaches an exact number. Not sure if that's a use case or if this would need to be allowed here, but I thought I raise the question :)

Reply:
Thanks for asking, this is why I need someone else thinking about it with me!

I think in that case we would want to return an actual integer but without a use case it's hard for me to think about how we would implement something like that....

Right now the only way the code implements an `UnInt` is through the uncertain time delta `UnDelta` object - there are more places I need to integrate it, so maybe as we use it more we'll have more clarity on what functionality we might need.